### PR TITLE
Crashlytics refactor report manager

### PR DESCRIPTION
**Pull request by mrwillis21**

### CODE REVIEW

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/SessionReportingCoordinator.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
  * This class handles Crashlytics lifecycle events and manages capture, persistence, and sending of
```
Comment (jakeouellette) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/SessionReportingCoordinator.java]: Should we use any language of "SessionReportingCoordinator"? e.g., "This Class coordinates Crashlytics lifecycle events and manages the capture, persistence, and sending of sessions as reports to Firebase Crashlytics"?

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/SessionReportingCoordinator.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
...
  * reports to Firebase Crashlytics.
  */
-public class FirebaseCrashlyticsReportManager implements CrashlyticsLifecycleEvents {
+public class SessionReportingCoordinator implements CrashlyticsLifecycleEvents {
```
Comment (jakeouellette) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/SessionReportingCoordinator.java]: Should we rename CrashlyticsLifecycleEvents to something less specific than SessionReportingCoordinator, e.g.,  SessionCoordinator, or maybe call the interface SessionReportingCoordinator and rename this class to something more specific about how it does it?

Comment (mrwillis21) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/SessionReportingCoordinator.java]: In the long term, that's the plan, but this is one step in that direction. We'll come back to this.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/SessionReportingCoordinator.java (hunk 1 above)
Comment (mrwillis21) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/SessionReportingCoordinator.java]: Good point.

Comment (mrwillis21) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/SessionReportingCoordinator.java]: Done.

**Pull request merged**